src/functionaltests/java/com/ericsson/ei/subscriptions/query/QueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,229 @@
+
+
+@Ignore
+@AutoConfigureMockMvc
+public class QueryAggregatedObjectsTestSteps extends FunctionalTestBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryAggregatedObjectsTestSteps.class);
+
+    
+    private static final String aggregatedObjJsonPath = "src/test/resources/AggregatedObject.json";
+    private static final String missedNotificationJsonPath = "src/test/resources/MissedNotification.json";
+
+    
+    @Autowired
+    private MockMvc mockMvc;
+    private MvcResult mvcResult;
+    
+    @LocalServerPort
+    private int port;
+    
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+    
+    @Value("${spring.data.mongodb.database}")
+    private String eiDatabseName;
+    
+    @Value("${aggregated.collection.name}")
+    private String aggrCollectionName;
+    
+    @Value("${missedNotificationCollectionName}")
+    private String missedNotificationCollectionName;
+    
+    @Value("${missedNotificationDataBaseName}")
+    private String missedNotificationDatabaseName;
+
+    private String aggrObj;
+    private String missedNotificationObj;
+    
+    private ObjectMapper objMapper;
+    
+    public QueryAggregatedObjectsTestSteps() {
+        objMapper = new ObjectMapper();
+
+        try {
+            aggrObj = FileUtils.readFileToString(new File(aggregatedObjJsonPath), "UTF-8");
+            missedNotificationObj = FileUtils.readFileToString(new File(missedNotificationJsonPath), "UTF-8");
+
+        } catch (IOException e) {
+            LOGGER.error("Failed to open test json files for test. Error message\n: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    @Given("^Aggregated object is created$")
+    public void aggregated_object_is_created() throws Throwable {
+        LOGGER.debug("Creating aggregated object in MongoDb");
+        assertEquals(true, createDocumentInMongoDb(eiDatabseName, aggrCollectionName, aggrObj));
+    }
+
+    @Given("^Missed Notification object is created$")
+    public void missed_notification_object_is_created() throws Throwable {
+        LOGGER.debug("Missed Notification object has been created in MongoDb");
+        assertEquals(true, createDocumentInMongoDb(missedNotificationDatabaseName, missedNotificationCollectionName, missedNotificationObj));
+    }
+
+    @Then("^Perform valid query on newly created Aggregated object")
+    public void perform_valid_query_on_newly_created_aggregated_object() throws Throwable {
+        String expectedTestCaseStartedEventId = "cb9d64b0-a6e9-4419-8b5d-a650c27c59ca";
+        String entryPoint = "/queryAggregatedObject";
+        String queryRequest = "{\"id\":\"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43\"}";
+        String documentId = getDocumentIdBasedOnQuery(eiDatabseName, aggrCollectionName, queryRequest);
+        LOGGER.debug("Got AggregateObject actual DocumentId after querying MongoDB: " + documentId);
+        mvcResult = mockMvc.perform(get(entryPoint)
```

Comment:
Yepp, I know I can moved this out to an help function, but I choose to keep the request inside each test due to it is not many lines and code becomes easier to follow.